src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,296 @@
+use std::io::{Result, Write};
+
+use analysis;
+use env::Env;
+use file_saver::save_to_file;
+use library::*;
+use library::Type as LType;
+use writer::primitives;
+use nameutil::*;
+
+use stripper_interface as stripper;
+use stripper_interface::Type as SType;
+use stripper_interface::TypeStruct;
+use stripper_interface::{FILE, /*FILE_COMMENT, */MOD_COMMENT};
+
+trait FunctionTraitType {
+    fn functions(&self) -> &Vec<Function>;
+    fn name(&self) -> &str;
+}
+
+macro_rules! impl_function_trait_type {
+    ($ty:ident) => {
+        impl FunctionTraitType for $ty {
+            fn functions(&self) -> &Vec<Function> {
+                &self.functions
+            }
+
+            fn name(&self) -> &str {
+                &self.name
+            }
+        }
+    }
+}
+
+impl_function_trait_type!(Class);
+impl_function_trait_type!(Interface);
+
+trait ToStripperType {
+    fn convert(&self) -> stripper::TypeStruct;
+    fn doc(&self) -> Option<String>;
+    fn doc_deprecated(&self) -> Option<String>;
+}
+
+macro_rules! impl_to_stripper_type {
+    ($ty:ident, $enum_var:ident) => {
+        impl ToStripperType for $ty {
+            fn convert(&self) -> TypeStruct {
+                TypeStruct::new(SType::$enum_var, &self.name)
+            }
+
+            fn doc(&self) -> Option<String> {
+                self.doc.clone()
+            }
+
+            fn doc_deprecated(&self) -> Option<String> {
+                self.doc_deprecated.clone()
+            }
+        }
+    }
+}
+
+impl_to_stripper_type!(Alias, Struct);
+impl_to_stripper_type!(Constant, Const);
+impl_to_stripper_type!(Member, Variant);
+impl_to_stripper_type!(Enumeration, Enum);
+impl_to_stripper_type!(Bitfield, Type);
+impl_to_stripper_type!(Record, Type);
+impl_to_stripper_type!(Field, Variant);
+impl_to_stripper_type!(Union, Struct);
+impl_to_stripper_type!(Function, Fn);
+impl_to_stripper_type!(Interface, Trait);
+impl_to_stripper_type!(Class, Struct);
+impl_to_stripper_type!(Namespace, Mod);
+impl_to_stripper_type!(Parameter, Variant);
+
+pub fn generate(env: &Env) {
+    let path =  env.config.target_path.join("doc.cmts");
+    println!("Generating documentation {:?}", path);
+    save_to_file(&path, env.config.make_backup,
+        |w| generate_doc(w, &env));
+}
+
+fn generate_doc(mut w: &mut Write, env: &Env) -> Result<()> {
+    //let namespace = lib.namespace(MAIN);
+
+    //try!(create_sub_docs(&mut output, &namespace.constants, None));
+    /*for ty in namespace.types.iter().filter_map(|t| t.as_ref()) {
+        try!(handle_type(&mut w, &ty, None));
+    }*/
```

Comment:
I exepct the enums and bitfields to be missing from configs for a while. But that could be discussed I guess.
In the meantime we'd still like to document them even if it's right in the sys crates where they're now.